src/lib.rs

```diff
@@ -0,0 +1,64 @@
+//#![cfg_attr(not(feature = "std"), no_std)]
+use ed25519_dalek::{VerifyingKey, Signature};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+
+#[wasm_bindgen]
+pub fn verify_ed25519(
+    data: &[u8], // [key (32 bytes), signature (64 bytes), message (? bytes); item_length]
+    item_length: usize,
+) -> Vec<u8> {
+    let results = data.chunks(item_length).map(|chunk| {
+        let maybe_key = &chunk[0..32].try_into();
+        let maybe_signature = &chunk[32..96].try_into();
+
+        if maybe_key.is_err() || maybe_signature.is_err() {
+            return 0;
+        }
+
+        let key = maybe_key.unwrap();
+        let signature = maybe_signature.unwrap();
```

Comment:
To avoid these `unwraps` you can do something like this:
```rust
let (key, signature) = match (maybe_key, maybe_signature) {
  (Ok(key), Ok(signature)) => (key, signature),
  _ => return 0;
};
```
or you can create a closure that will return `Result<(), SomeCommonError>` and then just do:
```rust
let closure = || {
 ....try_into()?;
};

match closure() {
  Ok(()) => 1,
  Err(_) => 0,
}
```

Reply:
fixed